schematic/src/i_callbacks.c

```diff
@@ -2459,52 +2459,7 @@ DEFINE_I_CALLBACK(attributes_attach)
   o_undo_savestate_old(w_current, UNDO_ALL);
 }
 
-/*! \todo Finish function documentation!!!
- *  \brief
- *  \par Function Description
- *
- */
-DEFINE_I_CALLBACK(attributes_detach)
-{
-  GschemToplevel *w_current = GSCHEM_TOPLEVEL (data);
-  GList *s_current;
-  OBJECT *o_current;
-  GList *detached_attribs = NULL;
-
-  g_return_if_fail (w_current != NULL);
 
-  /* This is a new addition 3/15 to prevent this from executing
-   * inside an action */
-  if (w_current->inside_action) {
-    return;
-  }
-
-  /* same note as above on i_update_middle_button */
-  i_update_middle_button(w_current, i_callback_attributes_detach,
```

Comment:
`(middle-button "repeat")` won't work any more with the
`attributes-detach` action since `i_update_middle_button()` is not
used in your Scheme replacement procedure.  I don't know how to
deal with it in Scheme just now :-/

Reply:
All that "repeat" stuff looks odd to me; probably, it should be somehow
connected with actions (there is already "repeat last action" command).
And mouse buttons indicators (IMHO) brings nothing but confusion and
complicates the code.

Reply:
My concern is mostly not about the GUI (BTW,  I cannot remember when and why the mouse button indication got removed from the status line, maybe you could recall?).  Our action code allows using dot (`.`) to repeat the last action. Can it somehow be used to do the same by the middle mouse button?

Reply:
Indicators weren't removed, but can be hidden by setting `show-mouse-buttons`
to `false` (`schematic.status-bar` group). They should be visible, by default.
`&repeat-last-action` can easily be invoked by the middle button; could you
please merge your fixes to the actions code first (#435)?
And I would remove `i_update_middle_button()` and associated code (it's
simply ugly) and display just "Repeat" in the middle mouse button indicator.
These indicators already takes up too much space in the status bar (fortunately,
they can be turned off now). I seriously doubt in their usefulness. We'd better
expand cryptic `Grid(100,100)` string.

Reply:
Middle mouse button works as expected.  Again, I didn't managed to prove it (I definitely need some rest :-)), I'm sorry.  OK, I'm going to merge #435 and this PR. Is the last code snippet about to fix the GUI issue we discussed above? If so, please make another PR and we'll finish with it.

Reply:
Middle button indicator will not be updated. The code above shows how
to make middle mouse button invoke "repeat last action" (`.`) command.
It could be done after removing `i_update_middle_button()` stuff and
making middle button indicator text static. Would you mind if I start this cleanup?

Reply:
Since I haven't understand how strong the first two sentences are related to the latter ones, I'll try to answer them apart:

> The code above shows how
to make middle mouse button invoke "repeat last action" (.) command.

I'm all for using high-level Scheme language instead of low-level C where possible or convenient. OTOH, you know, I'm not for changes for the sake of changes, I'm for convenience and productiveness...

> It could be done after removing i_update_middle_button() stuff and
making middle button indicator text static. Would you mind if I start this cleanup?

...Please go for it if you think things will be better with your changes (in using, coding, etc.) and please stop it, if not.

Reply:
Yes, this will free us from updating almost useless static indicators on each
and every action (cleaner, faster code), will allow to use high-level code
(`&repeat-last-action`) and it will make possible to move more code to
Scheme (I'm going to do so with `Attributes->Attach` next, which is also
not working well).